openmc/deplete/microxs.py

```diff
@@ -192,8 +209,93 @@ def __init__(self, data: np.ndarray, nuclides: List[str], reactions: List[str]):
+
+        Returns
+        -------
+        MicroXS
+        """
+        # check energies
+        if isinstance(energies, str):
+            energies = openmc.EnergyFilter.from_group_structure(energies).values
+        else:
+            # if user inputs energies check they are ascending (low to high) as some
+            # depletion codes use high energy to low energy.
+            if not all(energies[i] <= energies[i+1] for i in range(len(energies) - 1)):
+                raise ValueError('Energy bin must be in ascending order')
+
+        # check dimension consistency
+        if not len(multi_group_flux) == len(energies)-1:
+            raise ValueError('Length of flux array should be len(energies)-1')
+
+        chain_file_path = _resolve_chain_file_path(chain_file)
+        chain = openmc.deplete.Chain.from_xml(chain_file_path)
+
+        openmc_data_path = _resolve_openmc_data_path(openmc_data_path)
+        data_lib = openmc.data.DataLibrary.from_xml(openmc_data_path)
+
+        # get reactions and nuclides from chain file
+        nuclides, reactions = chain.nuclides, chain.reactions
+        nuclides = [nuc.name for nuc in nuclides]
+        if 'fission' in reactions:
+            # send to back
+            reactions.append(reactions.pop(reactions.index('fission')))
+        # convert reactions to mt file
+        #! I feel like this zero indexing will backfire
+        #! There has to be a better way for this
+        mts = [list(REACTIONS[reaction].mts)[0] for reaction in reactions if reaction != 'fission']
+        if 'fission' in reactions:
+            mts.append(18) # fission is not in the REACTIONS
+
+        # normalize flux and get energy midpoint
+        norm_multi_group_flux = np.array(multi_group_flux) / sum(multi_group_flux)
+        energies_midpoint = [(energies[i]+energies[i+1])/2 for i in range(len(energies)-1)]
+        temperature_key = f'{temperature}K'
+        microxs_arr = np.zeros((len(nuclides), len(mts)))
+
+        for nuc_indx, nuc in enumerate(nuclides):
+            mat_lib = data_lib.get_by_material(nuc, data_type='neutron')
+            if not mat_lib: # file does not exist
+                microxs_arr[nuc_indx, :] = 0
+                continue
+            else:
+                hdf5_path = mat_lib['path']
+            nuc_data = openmc.data.IncidentNeutron.from_hdf5(hdf5_path)
+            for mt_indx, mt in enumerate(mts):
+                try: # sometimes it fails cause there's no entry
+                    # perhaps a multigroup mgxs should be made as chosing the xs at the energy bin center is not always fair
+                    total_xs = np.array(nuc_data[mt].xs[temperature_key](energies_midpoint))
+                    # collapse
+                    total_norm = sum(total_xs * norm_multi_group_flux)
+                    microxs_arr[nuc_indx, mt_indx] = total_norm
+                except KeyError:
+                    microxs_arr[nuc_indx, mt_indx] = 0.0
```

Comment:
I believe the second value in `microxs_arr` needs to be an element of `_valid_rxns` -- which is just the `chain.REACTIONS` variable with the addition of `'fission'` -- NOT an MT number, so we'll need to convert from MT back to internal OpenMC reaction names.

Reply:
I might not be understanding properly, but it `mt_indx` should just be an integer right? For indexing the array? And that index should correspond to the `reactions`

Reply:
Ah, it seems I both misread and misunderstood what you were doing with the `microxs_arr`. You are absolutely correct.